### PR TITLE
[routing][traffic] Removed a circular dependency.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by bykoianko** [comment]
DUDE JTTP

**Comment by mpimenov** [comment]
@syershov The XCode part is up to you. I removed the model files from routing but I have not added the new routing_common library to the XCode project.

File: routing_common/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+project(routing_common)
+
+include_directories(
+  .
```
Comment (ygorshenin) [inline routing_common/CMakeLists.txt]: Why do we need this?

**Comment by bykoianko** [comment]
Не собирается под Android и iOS:

/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing/routing_helpers.hpp:25: error: undefined reference to 'routing::CarModel::AllLimitsInstance()'
/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing_common/vehicle_model.hpp:85: error: undefined reference to 'routing::VehicleModel::IsRoadType(unsigned int) const'
/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing/routing_helpers.hpp:26: error: undefined reference to 'routing::PedestrianModel::AllLimitsInstance()'
/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing_common/vehicle_model.hpp:85: error: undefined reference to 'routing::VehicleModel::IsRoadType(unsigned int) const'
/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing/routing_helpers.hpp:27: error: undefined reference to 'routing::BicycleModel::AllLimitsInstance()'
/Users/jenkins/root/workspace/GitHubWatchdog/BUILD_MODE/release/label/Android/omim/routing_common/vehicle_model.hpp:85: error: undefined reference to 'routing::VehicleModel::IsRoadType(unsigned int) const'
/Users/jenkins/android-ndk-r13b/sources/cxx-stl/llvm-libc++/include/memory:2157: error: undefined reference to 'routing::CarModelFactory::CarModelFactory()'
/Users/jenkins/android-ndk-r13b/sources/cxx-stl/llvm-libc++/include/memory:3792: error: undefined reference to 'vtable for routing::CarModelFactory'

File: routing_common/CMakeLists.txt (hunk 1 above)
Comment (mpimenov) [inline routing_common/CMakeLists.txt]: Removed.

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by mpimenov** [comment]
@bykoianko Android поправил, ios надо делать отдельно (обсуждали с @syershov).

**Comment by mpimenov** [comment]
PTAL

**Comment by mpimenov** [comment]
@bykoianko PTAL

**Comment by syershov** [comment]
DUDE JTTP

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
LGTM

**Comment by bykoianko** [comment]
LGTM

**Pull request merged**